src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/DeleteExtendedQueryTagService.cs

```diff
@@ -38,7 +73,37 @@ public async Task DeleteExtendedQueryTagAsync(string tagPath, CancellationToken
         }
 
         string normalizedPath = tags[0].GetPath();
-        ExtendedQueryTagStoreEntry extendedQueryTagEntry = await _extendedQueryTagStore.GetExtendedQueryTagAsync(normalizedPath, cancellationToken);
-        await _extendedQueryTagStore.DeleteExtendedQueryTagAsync(normalizedPath, extendedQueryTagEntry.VR, cancellationToken);
+
+        // Get tag and update status to deleting. If the key is not found or the status is already deleting, it will throw an exception.
+        ExtendedQueryTagStoreEntry xqt = await _extendedQueryTagStore.GetExtendedQueryTagAsync(normalizedPath, cancellationToken);
+        await _extendedQueryTagStore.UpdateExtendedQueryTagStatusToDelete(xqt.Key, cancellationToken);
+
+        // Start operation to delete extended query tag
+        Guid operationId = _guidFactory.Create();
+        await _client.StartDeleteExtendedQueryTagOperationAsync(operationId, xqt.Key, xqt.VR, cancellationToken);
+
+        // Wait for operation to finish to keep the API synchronous
+        OperationStatus finalStatus = await _retryPolicy.ExecuteAsync(async (ct) =>
+        {
+            IOperationState<DicomOperation> state = await _client.GetStateAsync(operationId, cancellationToken);
+
+            _logger.LogInformation(
+                "Waiting for the operation to delete extended query tag to complete. Status: {Status}",
+                state.Status);
+
+            return state.Status;
+        },
+        cancellationToken);
+
+        // If the operation is still in progress, or if it did not complete successfully, throw an exception
+        if (finalStatus.IsInProgress())
+        {
+            throw new DataStoreException($"The operation to delete extended query tags failed to complete in the alloted time. Status: {finalStatus}");
```

Comment:
This is getting the status of the operation. If we have waited for 15 minutes and not completed, the status would still be in progress. I chose to return an error here because something has probably gone wrong if the operation is still running after 15 minutes